src/components/ui/animated-button.tsx

```
@@ -0,0 +1,64 @@
+  ...props
+}: AnimatedButtonProps) {
+  const translateY = useSharedValue(0);
+
+  const animatedStyle = useAnimatedStyle(() => {
+    return {
+      transform: [{ translateY: translateY.value }],
+    };
+  });
+
+  const handlePressIn = (event: any) => {
+    translateY.value = withSpring(4, { damping: 50, stiffness: 500 });
```

ESLint

[error] Mutating a value returned from a function whose return value should not be mutated

(react-compiler/react-compiler)